bolt/lib/Core/DIEBuilder.cpp

```diff
@@ -897,6 +897,8 @@ void DIEBuilder::registerUnit(DWARFUnit &DU, bool NeedSort) {
                 });
   }
   getState().UnitIDMap[getHash(DU)] = getState().DUList.size();
+  if (getState().DUList.size() == getState().CloneUnitCtxMap.size())
```

Comment:
Please add a comment why this condition is the one for allocating a new CloneUnitCtxMap entry.